[custom_op] Rename register_impl to register_kernel

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124414
* #124403
* #124199
* #124134
* #124299
* __->__ #124200
* #124180

Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

zou3519 committed:
[custom_op] Rename register_impl to register_kernel

Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]

albanD reviewed (approved):
Yep, looks quite nice!

zou3519 committed:
Update on "[custom_op] Rename register_impl to register_kernel"


Motivation:
- The API is used for registering an implementation for a specific
  device type.
- "impl" is ambiguous and can be confused with Library.impl.

Test Plan:
- existing tests

[ghstack-poisoned]
